### PR TITLE
Fix "Metadata Search" tab toolTip in Data Source Manager dialog

### PR DESCRIPTION
**Pull request by DelazJ**
i.e. "Search and add layer metadata" instead of "Add Metadata Search layer"

### CODE REVIEW

**Comment by agiudiceandrea** [comment]
@DelazJ, I'm not sure how exactly the Metadata Search panel works, anyway I've found a [comment](https://github.com/qgis/QGIS/pull/50095#issuecomment-1238730545) by @nyalldawson to @elpaso that seems to clarify that:

> We're not adding metadata, we're adding layers by metadata

**Comment by DelazJ** [comment]
The [docs](https://docs.qgis.org/testing/en/docs/user_manual/managing_data_source/opening_data.html#the-layer-metadata-search-panel) written by @elpaso seems to indeed be about adding metadata but I might confess I never used it.
Let's wait for a more informed comment...

File: src/gui/qgslayermetadatasourceselectprovider.cpp [hunk 1]
```diff
@@ -34,6 +34,11 @@ QString QgsLayerMetadataSourceSelectProvider::text() const
   return QObject::tr( "Metadata Search" );
 }
 
+QString QgsLayerMetadataSourceSelectProvider::toolTip() const
+{
+  return QObject::tr( "Search and add layer metadata" );
```
Comment (nyalldawson) [inline src/gui/qgslayermetadatasourceselectprovider.cpp:39]: Maybe:

```suggestion
  return QObject::tr( "Find layers by metadata" );
```

Comment (DelazJ) [reply src/gui/qgslayermetadatasourceselectprovider.cpp:39]: @nyalldawson You are probably right... the only reference I have is unfortunately the misleading docs.

**Comment by DelazJ** [comment]
Well... I for some reason thought this was hard to setup but not at all, so I finally could play with the tab and can confirm it is about finding layers. Accordingly, there is a pr for updating the docs at https://github.com/qgis/QGIS-Documentation/pull/8461 I would appreciate feedback on. Thanks.

**Review by nyalldawson — APPROVED** [review]

**Pull request merged**